scripts/vsc-venv-wrapper/vsc-venv.sh

```diff
@@ -0,0 +1,182 @@
+  # use is_loaded_cluster_compatible_with_host
+
+  if ! is_loaded_cluster_compatible_with_host; then
+    local loaded_cluster="$VSC_INSTITUTE_CLUSTER"
+    local host_cluster="$VSC_DEFAULT_CLUSTER_MODULE"
+    echo_warning "The OS or architecture of the current host ($host_cluster) does not match that of the loaded cluster ($loaded_cluster)."
+    echo_warning "Creating or activating a virtual environment on $host_cluster with modules optimized for $loaded_cluster may cause issues."
+    echo_warning "to make a virtual environment on the $loaded_cluster cluster, start an interactive session on that cluster:"
+    echo_warning "  $ module swap cluster/$loaded_cluster"
+    echo_warning "  $ qsub -I"
+    echo_warning "  $ cd /path/to/your/project"
+    echo_warning "Then run the activate command again."
+    echo_warning "Exiting..."
+    return 1
+  fi
+
+  # === Step 1: Warn user if they have modules loaded === #
+
+  loaded_modules=($(echo "$LOADEDMODULES" | tr ':' '\n' | grep -v -E '^(env|cluster)/')) # Remove env and cluster modules
+  n_loaded_modules="${#loaded_modules[@]}"
+  if [ "$n_loaded_modules" -gt 0 ]; then
+    echo_warning "You have $n_loaded_modules loaded modules in the current shell. These modules will be purged."
+    echo_warning "If you want to use these modules, please provide a modules file listing the required modules as the second argument."
+
+    echo_warning "Loaded modules:"
+    for module in "${loaded_modules[@]}"; do
+      echo_warning "  $module"
+    done
+  fi
+
+
+  # === Step 2: Purge Modules === #
+
+  echo_info "Purging currently loaded modules."
+  module purge
+
+  # === Step 3: Load Modules if module script present === #
+
+  if [ -n "$modules_file" ]; then # If module script not empty
+
+    echo_info "Loading modules from '$modules_file'"
+
+    if ! load_modules "$modules_file"; then # If the modules could not be loaded
+      echo_error "Could not load modules from '$modules_file'"
+      return 1
+    fi
+
+    echo_info "Modules loaded successfully"
+
+  else
+    echo_info "No module file provided. Proceeding without extra modules."
+  fi
+
+  # === Step 4: Create Virtual Environment if not yet present === #
+
+  if [ "$(which python)" = "/usr/bin/python" ]; then
+    echo_warning "System python used. Consider loading a specific python module through the modules file."
+  fi
+
+  python_version=$(python --version)
+  echo_info "Using $python_version to create virtual environment at $venv_location"
+  # Will automatically make the venvs folder and venv, does nothing if they already exist
+  if ! python -m venv "$venv_location"; then
+    echo_error "Could not create virtual environment"
+    return 1
+  fi
+
+  # === Step 5: Activate Virtual Environment === #
+
+  echo_info "Activating virtual environment"
+  source "$venv_location/bin/activate"
+
+  # === Step 6: Install Requirements === #
+
+  echo_info "Installing requirements from '$requirements_file'"
```

Comment:
Check missing here to ensure that `${requirements_file}` really exists (or maybe do it higher up, as early as possible)